webv2/table/add_column.go

```diff
@@ -39,3 +54,45 @@ func addColumn(tableId string, colId string, conv *internal.Conv) {
 
 	conv.SpSchema[tableId] = sp
 }
+
+func AddNewColumn(w http.ResponseWriter, r *http.Request) {
+	log.Println("request started", "method", r.Method, "path", r.URL.Path)
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Println("request's body Read Error")
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+	}
+	tableId := r.FormValue("table")
+	details := columnDetails{}
+	err = json.Unmarshal(reqBody, &details)
+	if err != nil {
+		log.Println("request's Body parse error")
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+
+	sessionState := session.GetSessionState()
+	for _, c := range sessionState.Conv.SpSchema[tableId].ColDefs {
+		if strings.EqualFold(c.Name, details.Name) {
+			http.Error(w, fmt.Sprintf("Multiple columns with similar name cannot exist for column : %v", details.Name), http.StatusBadRequest)
+			return
+		}
+	}
+	usedNames := internal.ComputeUsedNames(sessionState.Conv)
+	_, found := usedNames[strings.ToLower(details.Name)]
+	if found {
+		http.Error(w, fmt.Sprintf("Name '%v' is in used names, please use a different column name", details.Name), http.StatusBadRequest)
+		return
```

Comment:
What is a "used name", the error message is not immediately clear I think.

Reply:
rephrased